rest/src/main/groovy/whelk/rest/api/ESQuery.groovy

```diff
@@ -0,0 +1,460 @@
+    }
+
+    /**
+     * Get sort clauses from query params, or null if not found.
+     *
+     * Public for test only - don't call outside this class!
+     *
+     */
+    public List getSortClauses(Map<String, String[]> queryParameters) {
+        if (!('_sort' in queryParameters)) return null
+        if (!(queryParameters.get('_sort').size() > 0)) return null
+
+        List result = []
+        String sortParams = queryParameters.get('_sort')[0]
+        sortParams.split(',').each { String sortParam ->
+            result << getSortClause(sortParam)
+        }
+        return result
+    }
+
+    @CompileStatic(TypeCheckingMode.SKIP)
+    private Map getSortClause(String sortParam) {
+        def (String field, String sortOrder) = getFieldAndSortOrder(sortParam)
+        String termPath = getInferredTermPath(field)
+        Map clause = [(termPath): ['order': sortOrder]]
+        if (field == 'hasTitle.mainTitle' || field == 'hasTitle.mainTitle.keyword') {
+            clause[termPath]['nested_path'] = 'hasTitle'
+            clause[termPath]['nested_filter'] = ['term': ['hasTitle.@type': 'Title']]
+        }
+        return clause
+    }
+
+    private Tuple2<String, String> getFieldAndSortOrder(String field) {
+        if (field.startsWith('-')) {
+            return new Tuple2(field.substring(1), 'desc')
+        } else {
+            return new Tuple2(field, 'asc')
+        }
+    }
+
+    private String getInferredTermPath(String termPath) {
+        if (termPath in keywordFields) {
+            return "${termPath}.keyword"
+        } else {
+            return termPath
+        }
+    }
+
+    /**
+     * Get site filter clause from query params, or null if not found.
+     *
+     * Public for test only - don't call outside this class!
+     *
+     */
+    public List getSiteFilter(Map<String, String[]> queryParameters) {
+        if (!('_site_base_uri' in queryParameters)) return null
+        if (!(queryParameters.get('_site_base_uri').size() > 0)) return null
+
+        String siteBaseUri = queryParameters.get('_site_base_uri')[0]
+        List prefixFilters = [
+            ['prefix': ['@id': siteBaseUri]],
+            ['prefix': ['sameAs.@id': siteBaseUri]]
+        ]
+
+        // We want either of the prefix filters to match, so we put them
+        // inside a bool query
+        return [['bool': ['should': prefixFilters]]]
+    }
+
+    /**
+     * Return a bool filter of any non-reserved query params, or null if no
+     * params found.
+     *
+     * Public for test only - don't call outside this class!
+     *
+     */
+    @CompileStatic(TypeCheckingMode.SKIP)
+    public List getFilters(Map<String, String[]> queryParameters) {
+        List filters = []
+
+        // TODO This is copied from the old code and should be rewritten.
+        // We don't have that many nested mappings, so this is way to general.
+        Map groups = queryParameters.groupBy { p -> getPrefixIfExists(p.key) }
```

Comment:
For simple lambdas, prefer either a name or `it` over a single letter. Both `it` and a single letter requires the reader to understand the thing being operated upon (which IMHO is good for comprehension), and (IMHO) `it` makes that (fairly) obvious.

Reply:
I agree! As discussed afk, we'll keep it as is for now.